### PR TITLE
Blog: PortSwigger Password Brute Force

### PR DESCRIPTION
**Pull request by psiinon**
Submitted on behalf of jan-ambroz

@jan-ambroz I've added an intro and tweaked a couple of things - can you check you are still happy with it?
Also, can you confirm that the user doesnt need to actually change wiener's password after having found it? 

Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

**Comment by psiinon** [comment]
Oops, nearly forgot the screenshot ;)

Screenshot 2022-03-28 at 11-29-47 PortSwigger Labs Password Brute-force via Password Change with ZAP

File: site/content/blog/2022-03-28-portswigger-lab-brute-force-password-change/index.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+but not too surprisingly it is also intended as a showcase for Burp Suite.
+
+However you can of course use ZAP instead of Burp, even those labs which require Burp Pro.
+
+In this series of ZAP Blog posts I will show you how.
+
+The first lab has the tag PRACTITIONER and you can find it [here](https://portswigger.net/web-security/authentication/other-mechanisms/lab-password-brute-force-via-password-change). 
+I want to show you how to solve this lab using ZAP and specifically the [Fuzzer](/docs/desktop/addons/fuzzer/) tool.
+
+### Lab instructions:
```
Comment (kingthorin) [inline site/content/blog/2022-03-28-portswigger-lab-brute-force-password-change/index.md:30]:
```suggestion
### Lab instructions
```

File: site/content/blog/2022-03-28-portswigger-lab-brute-force-password-change/index.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+This lab's password change functionality makes it vulnerable to brute-force attacks. To solve the lab, use the list of candidate passwords to brute-force Carlos's account and access his "My account" page.
```
Comment (kingthorin) [inline site/content/blog/2022-03-28-portswigger-lab-brute-force-password-change/index.md:32]:
```suggestion
This lab's password change functionality makes it vulnerable to brute-force attacks. To solve the lab, use the list of candidate passwords to brute-force Carlos' account and access his "My account" page.
```

File: site/content/blog/2022-03-28-portswigger-lab-brute-force-password-change/index.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+When an attack is done you can notice that one message has a different __Size  Resp. Body__ than the rest of the messages.
+
+![Fuzzer Response Size](images/fuzzer_response_size.png)
+
+With a closer look we can see the response:
+
+![New Password Does Not Match](images/response_password_does_not.png)
+
+When you now try to login to your account with the new credentials lab will be solved :smile:.
```
Comment (kingthorin) [inline site/content/blog/2022-03-28-portswigger-lab-brute-force-password-change/index.md:81]:
```suggestion
When you now try to login to your account with the new credentials the lab will be solved :smile:.
```

**Comment by psiinon** [comment]
Thanks, fixed those :)

**Comment by jan-ambroz** [comment]
> Submitted on behalf of jan-ambroz
> 
> @jan-ambroz I've added an intro and tweaked a couple of things - can you check you are still happy with it? Also, can you confirm that the user doesnt need to actually change wiener's password after having found it?
> 
> Signed-off-by: Simon Bennetts [psiinon@gmail.com](mailto:psiinon@gmail.com)
@psiinon it looks really good! 
no, we don't need to change the wiener's password.

**Comment by psiinon** [comment]
Updated to todays date, no longer WIP :)

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you both!

**Pull request merged**